### PR TITLE
[CARBONDATA-3889] Enable scala check style

### PR DESCRIPTION
**Pull request by QiangCai**
### Why is this PR needed?
1. scala style checking is not working during compile and IntelliJ IDEA does not use the scala style config in the dev folder.
2. Intellij IDEA sort import with different order
 
 ### What changes were proposed in this PR?
 1. move scalastyle-config.xml to the parent project folder.
    a) IntelliJ IDEA will show the style error during coding.
    b) Fix scala style issue which the scala style checking report during maven compile.
 2. change java-code-format-template.xml to sort import as Scalastyle. 

 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - No

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3720/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1981/

File: pom.xml [hunk 1]
```diff
@@ -431,9 +431,10 @@
           <failOnWarning>false</failOnWarning>
           <sourceDirectory>${basedir}/src/main/scala</sourceDirectory>
           <testSourceDirectory>${basedir}/src/test/scala</testSourceDirectory>
-          <configLocation>${dev.path}/scalastyle-config.xml</configLocation>
+          <configLocation>scalastyle-config.xml</configLocation>
```
Comment (xubo245) [inline pom.xml:434]: why. do you remove ${dev.path}/?

Comment (QiangCai) [reply pom.xml:434]: moved scalastyle-config.xml into the parent project folder.

**Comment by xubo245** [comment]
LGTM

**Comment by xubo245** [comment]
LGTM

**Comment by xubo245** [comment]
LGTM

**Comment by xubo245** [comment]
Thanks for your contribution!

**Comment by xubo245** [comment]
Thanks for your contribution!

**Pull request closed without merging**